File: lintcode/python/0150_best_time_to_buy_and_sell_stock_II.py
#!/usr/bin/python -t

# greedy time O(n) space O(1)

class Solution:
    """
    @param prices: Given an integer array
    @return: Maximum profit
    """
    def maxProfit(self, prices):
        # write your code here
        diff = profit = 0
        n = len(prices)
        for i in range(1, n):
            diff = prices[i] - prices[i-1]
            if diff > 0:
                profit += diff
                
        return profit

File: lintcode/golang/0150_best_time_to_buy_and_sell_stock_II.go
/**
 * @param prices: Given an integer array
 * @return: Maximum profit
 */
func maxProfit (prices []int) int {
    // write your code here
    ret := 0
    n := len(prices)
    if n == 0 {
        return ret
    }
    
    for i := 1; i < n; i++ {
        diff := prices[i] - prices[i-1]
        if diff > 0 {
            ret += diff
        }
    }
    
    return ret
}

